src/Grid/grid.jl

```diff
@@ -1450,3 +1605,76 @@ struct SurfaceOrientationInfo
     #flipped::Bool
     #shift_index::Int
 end
+# This one is the same as `SurfaceOrientationInfo` but used in interfaces. It is calculated for the other facet relative to current facet
+struct InterfaceOrientationInfo
+    flipped::Bool
+    transformation::Union{Nothing, AbstractArray}
+end
+
+function transform_coordinates(src::Vector{Vector{Float64}}, dst::Vector{Vector{Float64}})
+    # Calculate the transformation matrix
+    src_matrix = [  src[1][1]   src[2][1]   src[3][1]
+                    src[1][2]   src[2][2]   src[3][2]
+                    1           1           1       ]
+    dst_matrix = [  dst[1][1]   dst[2][1]   dst[3][1]
+                    dst[1][2]   dst[2][2]   dst[3][2]
+                    1           1           1       ]
+    transform_matrix = dst_matrix / src_matrix
+
+    return transform_matrix
+end
+# TODO: make this `SurfaceOrientationInfo`?
+"""
+    InterfaceOrientationInfo(grid::AbstractGrid, this_face::FaceIndex, other_face::FaceIndex)
+
+Orientation information for interfaces. Such an interface can be 
+possibly flipped (i.e. the defining vertex order is reverse to the 
+spanning vertex order) and the vertices can be rotated skewed against each other, and skewed in the case of triangles.
+Take for example the faces
+```
+1           3
+| \\         | \\
+|  \\        |  \\
+| A \\       | B \\ 
+|    \\      |    \\
+2-----3     1-----2  
+```
+which are skewed and rotated against each other by 90° or the faces
+```
+1           3
+| \\         | \\
+|  \\        |  \\
+| A \\       | B \\ 
+|    \\      |    \\
+2-----3     2-----1  
+```
+which are flipped about the centroid vector. Any combination of these can happen. 
+The combination to map other facet to current facet is encoded with
+this data structure via ``\\vec{p}_{\\text{there}} = [T] \\vec{p}_{\\text{here}}`` where ``[T]`` is `InterfaceOrientationInfo.transformation`.
+"""
+function InterfaceOrientationInfo(grid::AbstractGrid, this_face::FaceIndex, other_face::FaceIndex)
```

Comment:
We should be able to provide an optimized constructor by also providing a topology.

Reply:
I'm not sure how we'd use topology here, `InterfaceOrientationInfo` will be calculated inside `InterfaceIterator` which provides `this_face::FaceIndex, other_face::FaceIndex` from topology, I mean we only need these two from topology and if we pass topology we'd still need to pass `this_face::FaceIndex` unless we're buffering it for all interfaces If I understand correctly.

Reply:
Ah sorry, I misread here. You are right.